sdk/python/feast/infra/offline_stores/remote.py

```diff
@@ -121,23 +127,28 @@ def get_historical_features(
     ) -> RemoteRetrievalJob:
         assert isinstance(config.offline_store, RemoteOfflineStoreConfig)
 
-        # TODO: extend RemoteRetrievalJob API with all method parameters
-
         # Initialize the client connection
         location = f"grpc://{config.offline_store.host}:{config.offline_store.port}"
         client = fl.connect(location=location)
         logger.info(f"Connecting FlightClient at {location}")
 
         feature_view_names = [fv.name for fv in feature_views]
         name_aliases = [fv.projection.name_alias for fv in feature_views]
+
+        api_parameters = {
+            "feature_view_names": feature_view_names,
+            "feature_refs": feature_refs,
+            "project": project,
+            "full_feature_names": full_feature_names,
+            "name_aliases": name_aliases,
+        }
+
         return RemoteRetrievalJob(
             client=client,
-            feature_view_names=feature_view_names,
-            name_aliases=name_aliases,
-            feature_refs=feature_refs,
+            api="get_historical_features",
```

Comment:
If you want to avoid hardcoding these method names, you can use:
```py
api=OfflineStore.get_historical_features.__name__
```
Same for all the other APIs